### PR TITLE
feat: article key facts

### PR DESCRIPTION
**Pull request by colincclark**
https://nidigitalsolutions.jira.com/browse/REPLAT-1760

Adds key facts

### CODE REVIEW

File: packages/key-facts/src/key-facts.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+                        style={styles.link}
+                        url={url}
+                      >
+                        {renderedChildren}
+                      </TextLink>
+                    )
+                  };
+                }
+              },
+              `key-facts-${index}-${indx}`
```
Comment (craigbilner) [inline packages/key-facts/src/key-facts.js]: this is a little confusing?

File: packages/markup/__tests__/web/__snapshots__/markup-ad.web.test.js.snap [hunk 2]
```diff
@@ -23,7 +23,7 @@ exports[`multiple paragraphs with ads 1`] = `
     isLoading={false}
     section=""
     slotName="inline-ad"
-    style={85}
+    style={92}
```
Comment (craigbilner) [inline packages/markup/__tests__/web/__snapshots__/markup-ad.web.test.js.snap]: need to flatten styles

File: packages/styleguide/src/colours/functional.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 const functionalColours = {
+  bullet: "#103546",
```
Comment (craigbilner) [inline packages/styleguide/src/colours/functional.js]: this is specifically the key facts bullet?

File: packages/key-facts/fixtures/key-facts-no-title-showcase.json [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+              "attributes": {
+                "value": "Another ministerial aide, "
+              },
+              "children": []
+            },
+            {
+              "name": "link",
+              "attributes": {
+                "href":
+                  "https://www.thetimes.co.uk/edition/news/may-hit-by-another-brexit-walkout-amid-claims-of-establishment-coup-k9x69qlb8",
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/fixtures/key-facts-no-title-showcase.json]: this is the storybook fixture data - already added a cut down test fixture dataset

File: packages/styleguide/src/colours/functional.js (hunk 3 above)
Comment (colincclark) [reply packages/styleguide/src/colours/functional.js]: no, it is any bullet i think

File: packages/key-facts/README.md [hunk 5]
```diff
@@ -0,0 +1,38 @@
+# KeyFacts
+
+Unordered list of textual AST information that can contain text, links and
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/README.md]: amended

File: packages/key-facts/__tests__/android/__snapshots__/key-facts-with-style.android.test.js.snap [hunk 6]
```diff
@@ -0,0 +1,397 @@
...
+  "IS9": {
+    "flexDirection": "row",
+    "marginBottom": 20,
+    "paddingLeft": 1,
+    "width": "100%",
+  },
+}
+</style>
+
+<View>
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/__tests__/android/__snapshots__/key-facts-with-style.android.test.js.snap]: added

File: packages/key-facts/__tests__/android/__snapshots__/key-facts.android.test.js.snap [hunk 7]
```diff
@@ -0,0 +1,102 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`1. key facts with title 1`] = `
+<KeyFactsContainer>
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/__tests__/android/__snapshots__/key-facts.android.test.js.snap]: refactored

File: packages/key-facts/src/key-facts.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    <View key={`key-facts-${index}`} style={styles.container}>
+      <View style={styles.bulletContainer}>
+        <View style={styles.bullet} />
+        <Text style={styles.text}>
+          {item.children.map((data, indx) =>
+            renderTree(
+              data,
+              {
+                link(key, atts, renderedChildren) {
+                  const { canonicalId, href: url, type } = atts;
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/src/key-facts.js]: changed

File: packages/key-facts/src/key-facts.js (hunk 1 above)
Comment (colincclark) [reply packages/key-facts/src/key-facts.js]: amended

File: packages/markup/__tests__/web/__snapshots__/markup-ad.web.test.js.snap (hunk 2 above)
Comment (colincclark) [reply packages/markup/__tests__/web/__snapshots__/markup-ad.web.test.js.snap]: little out of scope, but have added the flattenStyle transformer. Not even sure why this changed

File: packages/key-facts/__tests__/ios/__snapshots__/key-facts-with-style.ios.test.js.snap [hunk 9]
```diff
@@ -0,0 +1,397 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
+exports[`1. key facts with title 1`] = `
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/__tests__/ios/__snapshots__/key-facts-with-style.ios.test.js.snap]: not sure what you mean here, but have removed the non-title test from the style-specific tests

File: packages/article/src/article-body/article-body-row.web.js [hunk 10]
```diff
@@ -98,6 +99,12 @@ const ArticleRow = ({ content: { data, index }, onLinkPress }) =>
         )
       };
     },
+    keyFacts(key, attributes, renderedChildren, indx, node) {
+      return {
+        element: <KeyFacts ast={node} key={key} onLinkPress={onLinkPress} />,
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/article/src/article-body/article-body-row.web.js]: it is passed down from `article` and attached to any links that are rendered using the bespoke KeyFacts render tree

File: packages/key-facts/key-facts.showcase.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+import { ScrollView } from "react-native";
+import { LateralSpacingDecorator } from "@times-components/storybook";
+import KeyFacts from "./src/key-facts";
+import data from "./fixtures/key-facts-showcase.json";
+import dataNoTitle from "./fixtures/key-facts-no-title-showcase.json";
+
+const renderKeyFacts = ast => <KeyFacts ast={ast} onLinkPress={() => {}} />;
+
+export default {
+  name: "Composed/Key Facts",
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/key-facts.showcase.js]: it is composed of things in the `renderTree`

File: packages/key-facts/package.json [hunk 12]
```diff
@@ -0,0 +1,74 @@
+{
+  "name": "@times-components/key-facts",
+  "version": "0.0.1",
+  "description": "Bulleted list of text data",
+  "main": "dist/key-facts",
+  "dev": "src/key-facts",
+  "scripts": {
+    "fmt": "prettier --write '**/*.*'",
```
[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/package.json]: we disucssed adding this to a separate PR? Will add

[the remark this answers is not in the document]
Comment (colincclark) [reply packages/key-facts/package.json]: done

File: packages/article/src/article-body/article-body-row.web.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (craigbilner) [reply packages/article/src/article-body/article-body-row.web.js]: indeed, but I don't believe we shold be using any `onPress` functionality on web which I've stared removing?

**Pull request merged**